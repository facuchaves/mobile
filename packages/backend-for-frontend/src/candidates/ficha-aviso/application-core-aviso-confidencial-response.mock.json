{
  "fechaPublicacion": "12-08-2019",
  "fechaFinalizacion": "11-09-2019",
  "estado": "activo",
  "titulo": "Analista Junior",
  "descripcion": "<p>Importante empresa busca Analistas Junior orientados a administracion para importantes proyectos a nivel nacional.</p><p> </p><p>Beneficios de comedor, transporte y descuento en importantes marcas.</p><p> </p><p> </p><p> </p>",
  "aptoDiscapacitados": false,
  "cantidadVacantes": 3,
  "empresa": {
    "denominacion": "Confidencial",
    "logoURL": null,
    "confidencial": true,
    "id": 13062978
  },
  "localizacion": {
    "paisId": 1,
    "zonaId": 1,
    "provinciaId": 29,
    "localidadId": 7,
    "direccion": null,
    "detalle": "Capital Federal, Buenos Aires, Argentina",
    "id": 7
  },
  "area": {
    "id": 1,
    "nombre": "Administración, Contabilidad y Finanzas"
  },
  "subArea": {
    "id": 22,
    "nombre": "Administración"
  },
  "requisitos": {
    "edad": null,
    "residencia": {
      "excluyente": false,
      "cercania": "pais",
      "cantidadKm": null
    },
    "experiencia": null,
    "genero": null,
    "educacion": null,
    "salario": null,
    "idiomas": null
  },
  "preguntas": [],
  "tipoTrabajo": {
    "id": 2,
    "nombre": "Full-time"
  },
  "planPublicacion": {
    "id": 10,
    "nombre": "Home"
  },
  "videoUrl": "www.youtube.com/watch?v=8UaZ9gHr0UQ",
  "nivelLaboral": {
    "nombre": "Junior",
    "id": 2
  },
  "redireccionURL": null,
  "origenAviso": "PORTAL",
  "puesto": {
    "puestoId": null,
    "nombre": null,
    "tieneEstadisticas": false
  },
  "id": 1113095924
}
