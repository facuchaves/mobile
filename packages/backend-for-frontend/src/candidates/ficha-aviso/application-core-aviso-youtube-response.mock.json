{
  "fechaPublicacion": "09-08-2019",
  "fechaFinalizacion": "08-09-2019",
  "estado": "activo",
  "titulo": "JEFE DE COMPRAS DE TEXTIL",
  "descripcion": "<p>Estamos en búsqueda de profesionales de las carreras de Comercialización, Marketing, Administración de Empresas y Diseño Textil para formar parte de nuestro equipo comercial.</p><p>La propuesta te permitirá llevar adelante la negociación con proveedores nacionales e internacionales, planificar las compras anuales, investigar las tendencias del mercado, desarrollar nuevos proveedores y realizar viajes al interior y exterior del país.</p><p>Valoraremos experiencia en puestos similares, preferentemente en Retail.</p><p> </p><p>Requisitos:</p><p>Son requisitos de la búsqueda, el nivel avanzado de Inglés, dominio de Excel y disponibilidad para realizar viajes al interior y exterior del país.</p><p> </p><p>Beneficios:</p><p>Nuestras oficinas están en la ciudad de Ituzaingó, Pcia. de Buenos Aires, y contamos con beneficios que te facilitarán el día a día:</p><p>Jornada flexible</p><p>Cochera</p><p>Comedor</p><p>Snacks y refrigerios saludables</p><p>Medicina prepaga</p><p>Descuentos en gimnasios y compras.</p><p> </p><p><br />Te invitamos a sumarte a nuestro equipo!</p><p> </p>",
  "aptoDiscapacitados": false,
  "cantidadVacantes": 1,
  "empresa": {
    "denominacion": "La Anónima",
    "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/0/0/0/0/6/2/4/0/3/0/logoMainPic_624030_bum_vd65f5702.jpg",
    "confidencial": false,
    "id": 624030
  },
  "localizacion": {
    "paisId": 1,
    "zonaId": 2,
    "provinciaId": 29,
    "localidadId": 296,
    "direccion": "Int. Pérez Quintana 3850",
    "detalle": "Ituzaingó, Buenos Aires, Argentina",
    "id": 15582351
  },
  "area": {
    "id": 6,
    "nombre": "Comercial, Ventas y Negocios"
  },
  "subArea": {
    "id": 23,
    "nombre": "Comercial"
  },
  "requisitos": {
    "edad": null,
    "residencia": {
      "excluyente": false,
      "cercania": "localidad",
      "cantidadKm": null
    },
    "experiencia": {
      "excluyente": true,
      "minimo": 36
    },
    "genero": null,
    "educacion": {
      "excluyente": true,
      "tipoEstudio": {
        "id": 3,
        "nombre": "Universitario"
      },
      "estadoEstudio": {
        "id": 2,
        "nombre": "Graduado"
      }
    },
    "salario": {
      "excluyente": false,
      "salarioMinimo": null,
      "salarioMaximo": null,
      "frecuenciaPago": null,
      "tipo": null,
      "solicitarCandidato": true
    },
    "idiomas": [
      {
        "excluyente": false,
        "idioma": {
          "id": 1,
          "nombre": "Inglés"
        },
        "nivel": {
          "id": 10,
          "nombre": "Avanzado"
        }
      }
    ]
  },
  "preguntas": [],
  "tipoTrabajo": {
    "id": 2,
    "nombre": "Full-time"
  },
  "planPublicacion": {
    "id": 10,
    "nombre": "Home"
  },
  "videoUrl": "www.youtube.com/watch?v=Ri2aMOoEnpM",
  "nivelLaboral": {
    "nombre": "Senior / Semi-Senior",
    "id": 1
  },
  "redireccionURL": null,
  "origenAviso": "HIRING_ROOM",
  "puesto": {
    "puestoId": 391,
    "nombre": "Jefe de Compras",
    "tieneEstadisticas": true
  },
  "id": 1113495614
}
