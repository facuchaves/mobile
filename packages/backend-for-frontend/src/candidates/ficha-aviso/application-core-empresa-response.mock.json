{
  "denominacion": "Navent",
  "razonSocial": "Navent SRL",
  "descripcion": "Somos la empresa multinacional líder en el mercado de Clasificados Online de Real Estate y Empleos. \nQueremos ayudar a que las personas logren dos de los anhelos más importantes de su vida: encontrar un empleo y un hogar. ",
  "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/0/0/1/0/1/1/3/7/7/5/logoMainPic_10113775_bum_v7d415042.jpg",
  "cuit": "27-29499968-5",
  "pais": {
    "id": 1,
    "nombre": "Argentina",
    "isoCode": "ar"
  },
  "provincia": {
    "id": 6,
    "nombre": "Capital Federal"
  },
  "ciudad": "CABA",
  "calle": "Echeverría ",
  "numero": "874",
  "piso": "",
  "departamento": "CABA",
  "codigoPostal": "-",
  "industria": {
    "id": 18,
    "nombre": "Internet"
  },
  "email": "empleos@navent.com",
  "webURL": null,
  "cantidadEmpleados": 800,
  "condicionIVA": {
    "nombre": "Responsable Inscripto",
    "id": 1
  },
  "consultora": false,
  "clienteCRMId": 10113775,
  "idUsuarioBumeranVendedor": null,
  "idClienteSap": "2000279113",
  "id": 10113775
}
