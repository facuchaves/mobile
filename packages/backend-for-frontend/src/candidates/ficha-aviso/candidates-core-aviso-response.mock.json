{
  "fechaPostulacion": "26-07-2019",
  "estadoPostulacion": "Estado mocked",
  "aviso": {
    "fechaPublicacion": "26-07-2019",
    "fechaFinalizacion": "25-08-2019",
    "estado": "activo",
    "titulo": "Analista BI Sr.",
    "descripcion": "<p>Somos la empresa multinacional líder en el mercado de Clasificados Online de Real Estate y Empleos. Queremos ayudar a que las personas logren dos de los anhelos más importantes de su vida: encontrar un empleo y un hogar. </p><p>Nuestra prioridad es que nuestros clientes tomen mejores decisiones, de manera más rápida y fácil. </p><p>Para nosotros siempre el usuario esta primero. </p><p>Las marcas que forman parte de Navent son: ZonaProp, Imovelweb, Wimoveis, Inmuebles24, Adondevivir, Plusvalia, Conlallave, CompreoAlquile, Union, CRM Inmobiliario, Bumeran, ZonaJobs, UniversoBit, Laborum, Multitrabajos,  Konzerta, Hiring Room entre otras.</p><p>¡En Navent estamos en búsqueda de Nuevos Talentos!</p><p>En esta oportunidad estamos en búsqueda de un <strong>Analista BI Sr</strong>. El objetivo del rol será el desarrollo de reportes y análisis de las operaciones, los cuales brindarán datos para la toma de decisiones. Realizar pedidos de análisis ad hoc del top management.<br /><br />Los requisitos que necesitamos para cubrir esta posición son:<br /><br /></p><ul><li>Graduado de las carreras de Ingeniería Industrial, Administración, Lic. En Economía, Adm. y sistemas.</li><li> </li><li>Experiencia previa de 3-5 años en posiciones similares.</li></ul><ul><li>Dominio de Office avanzado (Excluyente)</li></ul><ul><li>SQL avanzado (deseable)</li></ul><ul><li>Herramientas de Reporting y  Business Analytics (Datastudio)</li></ul><ul><li>Inglés intermedio (Deseable) / Portugués básico (Deseable)</li></ul><ul><li>Alta capacidad de análisis, buenas relaciones interpersonales,  y trabajo en equipo.</li></ul><p> </p><p>El lugar de trabajo es en nuestras oficinas que se encuentran en <strong>Belgrano</strong> y te ofrecemos como principales beneficios:</p><ul><li>Excelente clima laboral y oportunidad de desarrollo profesional</li></ul><ul><li>Obra Social incluyendo grupo familiar</li></ul><ul><li>Día de cumpleaños libre</li></ul><ul><li>Viernes Flex </li></ul><ul><li>Semana Navent (días libres entre las fiestas)</li></ul><ul><li>50% de descuento en Megatlon</li></ul><ul><li>Refrigerios</li></ul><p><br /><br /></p>",
    "aptoDiscapacitados": false,
    "cantidadVacantes": 1,
    "empresa": {
      "denominacion": "Navent",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/0/0/1/0/1/1/3/7/7/5/logoMainPic_10113775_bum_v7d415042.jpg",
      "confidencial": false,
      "id": 10113775
    },
    "localizacion": {
      "paisId": 1,
      "zonaId": 1,
      "provinciaId": 29,
      "localidadId": 7,
      "direccion": null,
      "detalle": "Capital Federal, Buenos Aires, Argentina",
      "id": 7
    },
    "area": {
      "id": 1,
      "nombre": "Administración, Contabilidad y Finanzas"
    },
    "subArea": {
      "id": 25,
      "nombre": "Finanzas"
    },
    "requisitos": {
      "edad": null,
      "residencia": {
        "excluyente": false,
        "cercania": "localidad",
        "cantidadKm": null
      },
      "experiencia": null,
      "genero": null,
      "educacion": null,
      "salario": null,
      "idiomas": null
    },
    "preguntas": [],
    "tipoTrabajo": {
      "id": 2,
      "nombre": "Full-time"
    },
    "planPublicacion": {
      "id": 40,
      "nombre": "Destacados"
    },
    "videoUrl": "vimeo.com/136332115",
    "nivelLaboral": {
      "nombre": "Senior / Semi-Senior",
      "id": 1
    },
    "redireccionURL": null,
    "origenAviso": "HIRING_ROOM",
    "puesto": {
      "puestoId": null,
      "nombre": null,
      "tieneEstadisticas": false
    },
    "id": 1113468691
  }
}
