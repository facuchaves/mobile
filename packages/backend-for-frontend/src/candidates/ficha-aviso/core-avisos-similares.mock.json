[
  {
    "titulo": "Analista BI Sr",
    "tipoTrabajo": "Full-time",
    "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/0/0/1/0/2/2/6/1/5/2/logoMainPic_10226152.jpg",
    "empresa": "Humana Partners",
    "localizacion": "Avellaneda, Buenos Aires",
    "fechaPublicacion": "19-07-2019",
    "salarioMinimo": 0,
    "salarioMaximo": 0,
    "aptoDiscapacitado": false,
    "planPublicacion": {
      "id": 1020,
      "nombre": "Aviso Talento"
    },
    "confidencial": false,
    "portal": "Bumeran",
    "oportunidad": false,
    "fechaHoraPublicacion": "19-07-2019 07:42:09",
    "idEmpresa": 10226152,
    "links": [
      {
        "rel": "self",
        "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113370082"
      }
    ],
    "id": 1113370082
  }
]
