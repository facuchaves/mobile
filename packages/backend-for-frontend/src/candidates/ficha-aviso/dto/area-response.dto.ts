import { ApiModelProperty } from '@nestjs/swagger'

export class AreaResponseDto {
  @ApiModelProperty({ required: false, description: 'id' })
  id?: number

  @ApiModelProperty({ required: false, description: 'nombre' })
  nombre?: string
}
