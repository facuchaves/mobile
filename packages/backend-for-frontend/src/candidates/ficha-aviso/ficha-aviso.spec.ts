import { FichaAvisoService } from './ficha-aviso.service'
import { RequestContext } from '../../models/request-context'
import { CandidatesCoreApiService } from '../candidates-core-api/candidates-core-api.service'
import { ApplicationCoreApiService } from '../../application-core-api/application-core-api.service'
import { ApplicationAvisosService } from '../../application-services/application-avisos.service'
import { FichaAvisoResponseDto } from '../../candidates/ficha-aviso/dto/ficha-aviso-response.dto'
import { RedisService } from '../../cache/cache-redis/redis.service'
import { AvisosService } from '../avisos/avisos.service'

describe('Ficha aviso', () => {
  afterEach(() => {
    jest.restoreAllMocks() // Se usa para los mocks de metodos estaticos que queda mockeado en todos los tests.
  })

  it('should get informacion solo de aviso usuario logeado', async () => {
    const candidatesCoreAvisosResponseMock = require('./candidates-core-aviso-response.mock.json')

    const candidatesCoreApiService = ({ get: () => {} } as unknown) as CandidatesCoreApiService
    const service = new FichaAvisoService(null, candidatesCoreApiService, null, null, null, null)

    const contextMocked = ({
      isCandidateLogged: () => true,
    } as unknown) as RequestContext

    jest
      .spyOn(candidatesCoreApiService, 'get')
      .mockImplementation(() => Promise.resolve(candidatesCoreAvisosResponseMock))

    const result = await service.getAviso(contextMocked, 1234)

    expect(candidatesCoreApiService.get).toHaveBeenCalledWith(contextMocked, `/v0/postulantes/avisos/1234`)

    expect(result).toMatchSnapshot()
  })

  it('should get informacion solo de aviso usuario no logeado', async () => {
    const applicationCoreAvisosResponseMock = require('./application-core-aviso-response.mock.json')

    const applicationCoreApiService = ({ get: () => {} } as unknown) as ApplicationCoreApiService

    const service = new FichaAvisoService(applicationCoreApiService, null, null, null, null, null)

    const contextMocked = ({
      isCandidateLogged: () => false,
    } as unknown) as RequestContext

    jest
      .spyOn(applicationCoreApiService, 'get')
      .mockImplementation(() => Promise.resolve(applicationCoreAvisosResponseMock))

    const result = await service.getAviso(contextMocked, 1234)

    expect(applicationCoreApiService.get).toHaveBeenCalledWith(contextMocked, `/v0/application/avisos/1234`)

    expect(result).toMatchSnapshot()
  })

  it('should get look and feel ok', async () => {
    const lookAndFeelMocked = require('./look-and-feel-response.mock.json')
    const empresaMocked = require('./application-core-empresa-response.mock.json')

    const applicationCoreApiService = ({ get: () => {} } as unknown) as ApplicationCoreApiService
    const applicationAvisosService = ({
      avisoLookAndFeel: (context, idAviso) => {
        lookAndFeelMocked
      },
    } as unknown) as ApplicationAvisosService

    const service = new FichaAvisoService(applicationCoreApiService, null, applicationAvisosService, null, null, null)

    const contextMocked = {} as RequestContext

    const avisoIdMock = 1234
    const empresaMock = {
      confidencial: false,
      denominacion: 'Denominacion empresa',
      id: 1,
    }

    jest
      .spyOn(applicationAvisosService, 'avisoLookAndFeel')
      .mockImplementation(() => Promise.resolve(lookAndFeelMocked))
    jest.spyOn(applicationCoreApiService, 'get').mockImplementation(() => Promise.resolve(empresaMocked))

    const productoLookAndFeel = await service.getProductoLookAndFeel(contextMocked, avisoIdMock, empresaMock)

    expect(applicationAvisosService.avisoLookAndFeel).toHaveBeenCalledWith(contextMocked, avisoIdMock)
    expect(applicationCoreApiService.get).toHaveBeenCalledWith(
      contextMocked,
      `/v0/application/empresas/empresas/${empresaMock.id}`
    )

    expect(productoLookAndFeel).toMatchSnapshot()
  })

  it('should get estadisticas ok', async () => {
    const estadisticasMocked = require('./estadisticas.mock.json')
    const redisService = ({ getValue: () => {} } as unknown) as RedisService
    const service = new FichaAvisoService(null, null, null, redisService, null, null)

    const contextMocked = ({
      site: {
        idPais: 1,
      },
    } as unknown) as RequestContext

    const avisoMock = ({
      aviso: {
        titulo: 'Titulo Aviso',
        id: 1234,
        puesto: {
          puestoId: 611,
          nombre: 'Nombre puesto mocked',
        },
        subArea: {
          nombre: 'Nombre area mocked',
        },
        empresa: {
          denominacion: 'Denominacion Empresa',
          id: 1,
        },
      },
    } as unknown) as FichaAvisoResponseDto

    jest.spyOn(redisService, 'getValue').mockImplementation(() => Promise.resolve(estadisticasMocked))

    const linkSeoSalario = await service.getEstadisticasPuesto(contextMocked, avisoMock)

    expect(linkSeoSalario).toMatchSnapshot()
  })

  it('should get avisos similares ok', async () => {
    const coreAvisosSimilaresMock = require('./core-avisos-similares.mock.json')
    const applicationCoreApiService = ({ get: () => {} } as unknown) as ApplicationCoreApiService
    const service = new FichaAvisoService(applicationCoreApiService, null, null, null, null, null)

    const contextMocked = ({} as unknown) as RequestContext

    const avisoIdMock = 1234

    jest.spyOn(applicationCoreApiService, 'get').mockImplementation(() => Promise.resolve(coreAvisosSimilaresMock))

    const avisosSimilares = await service.getAvisosSimilares(contextMocked, avisoIdMock)

    expect(applicationCoreApiService.get).toHaveBeenCalledWith(
      contextMocked,
      `/v0/application/avisos/${avisoIdMock}/similares?excluyeEmpresa=true`
    )

    expect(avisosSimilares).toMatchSnapshot()
  })

  it('should add video vimeo a aviso logueado', async () => {
    const candidatesCoreAvisosResponseMock = require('./candidates-core-aviso-response.mock.json')
    const service = new FichaAvisoService(null, null, null, null, null, null)

    const videoUrl = await service.getVideoUrl(candidatesCoreAvisosResponseMock.aviso.videoUrl)

    expect(videoUrl).toBeDefined()
    expect(videoUrl).toEqual('https://player.vimeo.com/video/136332115')
  })

  it('should add video youtube a aviso logueado', async () => {
    const applicationCoreAvisosResponseMock = require('./application-core-aviso-youtube-response.mock.json')
    const service = new FichaAvisoService(null, null, null, null, null, null)

    const videoUrl = await service.getVideoUrl(applicationCoreAvisosResponseMock.videoUrl)

    expect(videoUrl).toBeDefined()
    expect(videoUrl).toEqual('https://www.youtube.com/embed/Ri2aMOoEnpM?wmode=transparent')
  })

  it('should add seoFriendlyUrl a aviso logueado', async () => {
    const candidatesCoreAvisosResponseMock = require('./candidates-core-aviso-response.mock.json')
    const service = new FichaAvisoService(null, null, null, null, null, null)

    const { aviso } = candidatesCoreAvisosResponseMock
    const seoFriendlyUrl = await service.getSeoFriendlyUrl(aviso.titulo, aviso.empresa.denominacion, aviso.id)

    expect(seoFriendlyUrl).toBeDefined()
    expect(seoFriendlyUrl).toEqual('/empleos/analista-bi-sr.-navent-1113468691.html')
  })

  it('should ocultar info si es empresa confidencial', async () => {
    let empresaConfidencial = {
      denominacion: 'Confidencial',
      logoURL: null,
      confidencial: true,
      id: 13062978,
    }
    const service = new FichaAvisoService(null, null, null, null, null, null)

    await service.ocultarInformacionEmpresaConfidencial(empresaConfidencial)

    expect(empresaConfidencial).toBeDefined()
    expect(empresaConfidencial.confidencial).toBeDefined()
    expect(empresaConfidencial.confidencial).toBeTruthy()
    expect(empresaConfidencial.logoURL).toBe(undefined) //TODO Revisar esto
    expect(empresaConfidencial.id).toBe(undefined) //TODO Revisar esto
    expect(empresaConfidencial.denominacion).toBe(undefined) //TODO Revisar esto
  })

  it('should get avisos sugeridos', async () => {
    const candidatesCoreAvisosSugeridosMock = require('./candidates-core-avisos-sugeridos.mock.json')
    const avisosService = ({ search: () => {} } as unknown) as AvisosService

    const service = new FichaAvisoService(null, null, null, null, avisosService, null)

    jest.spyOn(avisosService, 'search').mockImplementation(() => Promise.resolve(candidatesCoreAvisosSugeridosMock))

    const result = await service.getAvisosSugeridos({} as RequestContext, 12, 34)

    expect(avisosService.search).toHaveBeenCalledWith({} as RequestContext, {
      filterData: { subAreasId: [34] },
    })

    expect(result).toBeDefined()
    expect(result.length).toBeGreaterThan(0)
    expect(result[0]).toBeDefined()
    expect(result[0].empresa).toBeDefined()
    expect(result[0].localizacion).toBeDefined()
    expect(result[0].link).toBeDefined()
    expect(result[0].titulo).toBeDefined()
    expect(result).toMatchSnapshot()
  })
})
