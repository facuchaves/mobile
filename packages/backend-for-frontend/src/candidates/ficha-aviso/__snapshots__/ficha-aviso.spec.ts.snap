// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ficha aviso should get avisos similares ok 1`] = `
Array [
  Object {
    "empresa": "Humana Partners",
    "fechaPublicacion": "19-07-2019 07:42:09",
    "link": "/empleos/analista-bi-sr-humana-partners-1113370082.html",
    "localizacion": "Avellaneda, Buenos Aires",
    "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/0/0/1/0/2/2/6/1/5/2/logoMainPic_10226152.jpg",
    "titulo": "Analista BI Sr",
  },
]
`;

exports[`Ficha aviso should get avisos sugeridos 1`] = `
Array [
  Object {
    "empresa": "Edenor",
    "link": "/empleos/aviso-semana-ingenieros-edenor-1113498023.html",
    "localizacion": "Capital Federal, Buenos Aires",
    "titulo": "Aviso Semana ingenieros",
  },
  Object {
    "empresa": "Edenor",
    "link": "/empleos/aviso-semana-ingenieros-edenor-1113498022.html",
    "localizacion": "Capital Federal, Buenos Aires",
    "titulo": "Aviso Semana ingenieros",
  },
  Object {
    "empresa": "Edenor",
    "link": "/empleos/aviso-semana-ingenieros-edenor-1113498021.html",
    "localizacion": "Capital Federal, Buenos Aires",
    "titulo": "Aviso Semana ingenieros",
  },
  Object {
    "empresa": "Edenor",
    "link": "/empleos/aviso-semana-ingenieros-edenor-1113498020.html",
    "localizacion": "Capital Federal, Buenos Aires",
    "titulo": "Aviso Semana ingenieros",
  },
  Object {
    "empresa": "Edenor",
    "link": "/empleos/aviso-semana-ingenieros-edenor-1113498019.html",
    "localizacion": "Capital Federal, Buenos Aires",
    "titulo": "Aviso Semana ingenieros",
  },
]
`;

exports[`Ficha aviso should get estadisticas ok 1`] = `"/salarios/nombre-area-mocked/nombre-puesto-mocked_611.html"`;

exports[`Ficha aviso should get informacion solo de aviso usuario logeado 1`] = `
Object {
  "aviso": Object {
    "aptoDiscapacitados": false,
    "area": Object {
      "id": 1,
      "nombre": "Administración, Contabilidad y Finanzas",
    },
    "cantidadVacantes": 1,
    "descripcion": "<p>Somos la empresa multinacional líder en el mercado de Clasificados Online de Real Estate y Empleos. Queremos ayudar a que las personas logren dos de los anhelos más importantes de su vida: encontrar un empleo y un hogar. </p><p>Nuestra prioridad es que nuestros clientes tomen mejores decisiones, de manera más rápida y fácil. </p><p>Para nosotros siempre el usuario esta primero. </p><p>Las marcas que forman parte de Navent son: ZonaProp, Imovelweb, Wimoveis, Inmuebles24, Adondevivir, Plusvalia, Conlallave, CompreoAlquile, Union, CRM Inmobiliario, Bumeran, ZonaJobs, UniversoBit, Laborum, Multitrabajos,  Konzerta, Hiring Room entre otras.</p><p>¡En Navent estamos en búsqueda de Nuevos Talentos!</p><p>En esta oportunidad estamos en búsqueda de un <strong>Analista BI Sr</strong>. El objetivo del rol será el desarrollo de reportes y análisis de las operaciones, los cuales brindarán datos para la toma de decisiones. Realizar pedidos de análisis ad hoc del top management.<br /><br />Los requisitos que necesitamos para cubrir esta posición son:<br /><br /></p><ul><li>Graduado de las carreras de Ingeniería Industrial, Administración, Lic. En Economía, Adm. y sistemas.</li><li> </li><li>Experiencia previa de 3-5 años en posiciones similares.</li></ul><ul><li>Dominio de Office avanzado (Excluyente)</li></ul><ul><li>SQL avanzado (deseable)</li></ul><ul><li>Herramientas de Reporting y  Business Analytics (Datastudio)</li></ul><ul><li>Inglés intermedio (Deseable) / Portugués básico (Deseable)</li></ul><ul><li>Alta capacidad de análisis, buenas relaciones interpersonales,  y trabajo en equipo.</li></ul><p> </p><p>El lugar de trabajo es en nuestras oficinas que se encuentran en <strong>Belgrano</strong> y te ofrecemos como principales beneficios:</p><ul><li>Excelente clima laboral y oportunidad de desarrollo profesional</li></ul><ul><li>Obra Social incluyendo grupo familiar</li></ul><ul><li>Día de cumpleaños libre</li></ul><ul><li>Viernes Flex </li></ul><ul><li>Semana Navent (días libres entre las fiestas)</li></ul><ul><li>50% de descuento en Megatlon</li></ul><ul><li>Refrigerios</li></ul><p><br /><br /></p>",
    "empresa": Object {
      "confidencial": false,
      "denominacion": "Navent",
      "id": 10113775,
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/0/0/1/0/1/1/3/7/7/5/logoMainPic_10113775_bum_v7d415042.jpg",
    },
    "estado": "activo",
    "fechaFinalizacion": "25-08-2019",
    "fechaPublicacion": "26-07-2019",
    "id": 1113468691,
    "localizacion": Object {
      "detalle": "Capital Federal, Buenos Aires, Argentina",
      "direccion": null,
      "id": 7,
      "localidadId": 7,
      "paisId": 1,
      "provinciaId": 29,
      "zonaId": 1,
    },
    "nivelLaboral": Object {
      "id": 1,
      "nombre": "Senior / Semi-Senior",
    },
    "origenAviso": "HIRING_ROOM",
    "planPublicacion": Object {
      "id": 40,
      "nombre": "Destacados",
    },
    "preguntas": Array [],
    "puesto": Object {
      "nombre": null,
      "puestoId": null,
      "tieneEstadisticas": false,
    },
    "redireccionURL": null,
    "requisitos": Object {
      "salario": false,
    },
    "subArea": Object {
      "id": 25,
      "nombre": "Finanzas",
    },
    "tipoTrabajo": Object {
      "id": 2,
      "nombre": "Full-time",
    },
    "titulo": "Analista BI Sr.",
    "videoUrl": "vimeo.com/136332115",
  },
  "postulacion": Object {
    "estado": "Estado mocked",
    "fecha": "26-07-2019",
  },
}
`;

exports[`Ficha aviso should get informacion solo de aviso usuario no logeado 1`] = `
Object {
  "aviso": Object {
    "aptoDiscapacitados": false,
    "area": Object {
      "id": 1,
      "nombre": "Administración, Contabilidad y Finanzas",
    },
    "cantidadVacantes": 1,
    "descripcion": "<p>Somos la empresa multinacional líder en el mercado de Clasificados Online de Real Estate y Empleos. Queremos ayudar a que las personas logren dos de los anhelos más importantes de su vida: encontrar un empleo y un hogar. </p><p>Nuestra prioridad es que nuestros clientes tomen mejores decisiones, de manera más rápida y fácil. </p><p>Para nosotros siempre el usuario esta primero. </p><p>Las marcas que forman parte de Navent son: ZonaProp, Imovelweb, Wimoveis, Inmuebles24, Adondevivir, Plusvalia, Conlallave, CompreoAlquile, Union, CRM Inmobiliario, Bumeran, ZonaJobs, UniversoBit, Laborum, Multitrabajos,  Konzerta, Hiring Room entre otras.</p><p>¡En Navent estamos en búsqueda de Nuevos Talentos!</p><p>En esta oportunidad estamos en búsqueda de un <strong>Analista BI Sr</strong>. El objetivo del rol será el desarrollo de reportes y análisis de las operaciones, los cuales brindarán datos para la toma de decisiones. Realizar pedidos de análisis ad hoc del top management.<br /><br />Los requisitos que necesitamos para cubrir esta posición son:<br /><br /></p><ul><li>Graduado de las carreras de Ingeniería Industrial, Administración, Lic. En Economía, Adm. y sistemas.</li><li> </li><li>Experiencia previa de 3-5 años en posiciones similares.</li></ul><ul><li>Dominio de Office avanzado (Excluyente)</li></ul><ul><li>SQL avanzado (deseable)</li></ul><ul><li>Herramientas de Reporting y  Business Analytics (Datastudio)</li></ul><ul><li>Inglés intermedio (Deseable) / Portugués básico (Deseable)</li></ul><ul><li>Alta capacidad de análisis, buenas relaciones interpersonales,  y trabajo en equipo.</li></ul><p> </p><p>El lugar de trabajo es en nuestras oficinas que se encuentran en <strong>Belgrano</strong> y te ofrecemos como principales beneficios:</p><ul><li>Excelente clima laboral y oportunidad de desarrollo profesional</li></ul><ul><li>Obra Social incluyendo grupo familiar</li></ul><ul><li>Día de cumpleaños libre</li></ul><ul><li>Viernes Flex </li></ul><ul><li>Semana Navent (días libres entre las fiestas)</li></ul><ul><li>50% de descuento en Megatlon</li></ul><ul><li>Refrigerios</li></ul><p><br /><br /></p>",
    "empresa": Object {
      "confidencial": false,
      "denominacion": "Navent",
      "id": 10113775,
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/0/0/1/0/1/1/3/7/7/5/logoMainPic_10113775_bum_v7d415042.jpg",
    },
    "estado": "activo",
    "fechaFinalizacion": "25-08-2019",
    "fechaPublicacion": "26-07-2019",
    "id": 1113468691,
    "localizacion": Object {
      "detalle": "Capital Federal, Buenos Aires, Argentina",
      "direccion": null,
      "id": 7,
      "localidadId": 7,
      "paisId": 1,
      "provinciaId": 29,
      "zonaId": 1,
    },
    "nivelLaboral": Object {
      "id": 1,
      "nombre": "Senior / Semi-Senior",
    },
    "origenAviso": "HIRING_ROOM",
    "planPublicacion": Object {
      "id": 40,
      "nombre": "Destacados",
    },
    "preguntas": Array [],
    "puesto": Object {
      "nombre": null,
      "puestoId": null,
      "tieneEstadisticas": false,
    },
    "redireccionURL": null,
    "subArea": Object {
      "id": 25,
      "nombre": "Finanzas",
    },
    "tipoTrabajo": Object {
      "id": 2,
      "nombre": "Full-time",
    },
    "titulo": "Analista BI Sr.",
    "videoUrl": "vimeo.com/136332115",
  },
}
`;

exports[`Ficha aviso should get look and feel ok 1`] = `
Object {
  "bannerEmpresa": "http://imgbum.prepro.jobscdn.com/postulantes/custom/bg_10113775.png",
  "colorBackgroundBotones": "#562567",
  "colorBackgroundCaja": "#000000",
  "colorBackgroundSeparadores": "#000000",
  "colorBackgroundTituloCaja": "#762b85",
  "colorBordeCaja": "#562567",
  "colorLabels": "#000000",
  "colorSeparadoresCaja": "#000000",
  "colorTexto": "#000000",
  "colorTextoBotones": "#FFFFFF",
  "colorTextoCaja": "#000000",
  "colorTextoSeparadores": "#000000",
  "colorTextoTituloCaja": "#FFFFFF",
  "colorTituloAviso": "#000000",
  "colorTituloEmpresa": "#000000",
  "empresa": Object {
    "descripcion": "Somos la empresa multinacional líder en el mercado de Clasificados Online de Real Estate y Empleos. 
Queremos ayudar a que las personas logren dos de los anhelos más importantes de su vida: encontrar un empleo y un hogar. ",
    "industria": "Internet",
    "nombre": "Navent",
  },
}
`;
